### DISCUSSION TITLE
Licensing questions

### DISCUSSION THREAD

**Discussion by lroal** [prisma/prisma#24290] [Q&A] [answered]
Hello, I am the maintainer of another ORM, https://gitHub.com/alfateam/orange-orm .
I am considering using prisma as the migration tool as Orange ORM does not have any. 
I think there are two alternatives here. 

1. Generate prisma file from Orange ORM and let the consumer handle interaction with prisma migration him/herself. 

2. Interact with prisma migration tool directly from Orange ORM not making it transient that it uses prisma -  other than having to install prisma manually (npm install).  

I was wondering if number 2 is violating  the license terms.

**Comment by janpio** [comment, marked as answer]
Hey @lroal, Prisma uses Apache 2.0, so pretty much anything goes: https://github.com/prisma/prisma?tab=Apache-2.0-1-ov-file#readme We're excited that you see Prisma as a good option for migrations.

Generating the Prisma schema is definitely the faster approach, but integration with our migration tooling under the hood can possibly create a better DX. I am not sure how easy it will be to adapt what you already have to talk to Prisma though, so if I was you I would make sure to use the correct abstraction level.

Let us know if you run into any trouble, and we'll try to help.

**Reply by lroal** [reply]
Thank's a lot 🙏😊
